ttkwidgets/shell.py

```diff
@@ -0,0 +1,205 @@
+        """
+        self.textvariable.set(self.buffer)
+        if self.last_text:
+            self.delete(self.last_text)
+        kwargs = {
+            'anchor': tk.NW,
+            'fill': 'white',
+            'text': self.buffer,
+            'width': self['width'],
+            'font': self.font,
+        }
+        self.last_text = self.create_text(*self.line_pos, **kwargs)
+        return
+
+    def cursor_blink(self):
+        if self.cursor:
+            self.delete(self.cursor)
+            self.cursor = None
+        if self.last_text is not None and self.blink:
+            pos = self.cursor_pos
+            font = self.itemcget(self.last_text, 'font').split()
+            width, height = self.max_char_width, int(font[-1])
+            pos = pos + (pos[0] + width, pos[1] + height)
+            self.cursor = self.create_rectangle(*pos, fill='white')
+        self.blink = not self.blink
+        self.after(1000, self.cursor_blink)
+
+    @property
+    def max_char_width(self):
+        """
+        Gets the width of a W character
+
+        :returns: width of W with the font the shell uses
+        :rtype: int
+        """
+        font = tkfont.Font(self, font=self.itemcget(self.last_text, 'font'))
+        return font.measure('W')
+
+    @property
+    def cursor_pos(self):
+        text = self.itemcget(self.last_text, 'text')
+        font = tkfont.Font(self, font=self.itemcget(self.last_text, 'font'))
+        text_len = font.measure(text)
+        width = self.max_char_width
+        span = self.text_line_span()
+        x = text_len % int(self['width']) + width + self.line_pos[0]
+        y = self.line_pos[1] + 15 * (span - 1)
+        return x, y
+
+    def text_line_span(self):
+        """
+        Gets the number of lines to display the current text
+
+        :returns: number of lines
+        :rtype: int
+        """
+        text = self.itemcget(self.last_text, 'text')
+        font = tkfont.Font(self, font=self.itemcget(self.last_text, 'font'))
+        text_len = font.measure(text)
+        n_lines = text_len // int(self['width']) + 1
+        return n_lines
+
+    def call_command(self, command, *args):
+        """
+        Calls command callbacks
+
+        :param command: Command name to call
+        :type command: str
+        """
+        commands = self.commands.get(command, [])
+        if callable(commands):
+            return commands(*args)
+
+        for callback in commands:
+            callback(*args)
+
+    def add_command(self, command, *callbacks):
+        """
+        Adds a command callback
+
+        :param command: command name to bind the callback to
+        :type command: str
+        :param *callbacks: list of callbacks to bind to the command
+        :type *callbacks: list[callable]
+        """
+        assert all(callable(callback) for callback in callbacks), f'Callback should be a function'
+        self.commands[command].extend(callbacks)
+
+    def print(self, message):
```

Comment:
Personally, I would change this function signature to  `print(self, *message)` to more closely match the behaviour of the `print` function. So something like:
```python
def print(self, *message):
    message =  " ".join(str(x) for x in message)
    self.buffer = message
    ...
```